### ISSUE TITLE
Validity of the form is not exposed in any way.

### ISSUE THREAD

**Issue by jporley** [makinacorpus/angular2-schema-form#77]
**Improvement:** Given that a developer might want to access the validity of the form as a whole, without having to submit the form, I think it would be a huge plus to provide another output for the form component.

Something like a `(onErrorsChange)="myHandler($event)"`

With this, just adding one property and a line of code (copy the behavior of onChanges), we would enable the user to receive the errors that exist in the form, or the absence of them. 

EDIT: I'd be happy to provide a PR for this during my free time :)

**Comment by TimjeKnight** [comment]
I managed a little workaround inside an action button ->

```
     {
        'submit': (form, options) => {
            // check if valid. if so, continue
            let formValid = true;
            form.forEachChildRecursive(child => {
                if (!child.valid) formValid = false;
            });
            if (formValid) {
                this.next();
            }
     }
```

**Comment by deviantdes** [comment]
hey , i agreed with this suggestion!!
any update on this?

**Comment by fbessou** [comment]
I agree `(onErrorsChange)="myHandler($event)"` would be the way to do.
If someone wants to implement it and make a PR before feel free !
I am very busy this week, but I can take a look next week if needed.

**Comment by ebrehault** [comment]
@fbessou, initially @jporley offered to make a PR, but he seems busy, so sure go ahead :)

**Comment by HenriPodolski** [comment]
Is this doable by just exposing errorsChange from rootProperty? I created a PR, that you know what I mean: https://github.com/makinacorpus/angular2-schema-form/pull/105

**Comment by thereiskeks** [comment]
any update on this?

**Comment by ebrehault** [comment]
Fixed by #123
